packages/typespec-vscode/test/manual/generate-client-code-from-typespec-test-plan.md

```diff
@@ -0,0 +1,129 @@
+# Generate Client Code from TypeSpec Test Scenario
+
+Generate client code from TypeSpec. In VS Code extension, we can complete code generation with step-by-step guidance.  
+TypeSpec Extension support will be extended to client code generation for first-class languages: `.NET`, `Python`, `Java`, and `JavaScript`.
+
+**Important: There must be at least one TypeSpec project in the project folder.**
+
+## Test Environment
+
+* OS : Windows or Linux
+* Language : Python, Java, JavaScript, .NET
+
+> Note: The extension should support all test cases in VS Code for Windows and Linux. Mac support is a stretch goal for Selenium semester.
+
+## Prerequisites
+
+Install TypeSpec Compiler before starting to write TypeSpec.
+
+* [Node.js 20+](https://nodejs.org/download/)
+* Npm 7+
+* [Install TypeSpec Compiler CLI](https://typespec.io/docs/): `"npm install -g @typespec/compiler"`
+
+Install required SDK/runtime for executing the specified language:
+
+* [.NET 8.0 SDK](https://dotnet.microsoft.com/en-us/download)
+* [Java 11 or above](https://www.oracle.com/java/technologies/downloads/), and [Maven](https://maven.apache.org/download.cgi)
+* [Python 3.8+](https://www.python.org/downloads/)
+* [Node.js 20+](https://nodejs.org/download/)
+
+## Test Steps
+
+### Step 1: Install the typespec extension.
+
+_Option 1_. Install using .vsix file: 
+   `Extension` -> `…` -> `Install form VSIX...`
+   
+   ![alt text](./images/InstallTypespec_VSIX.png)
+
+   Find the .vsix file you want to install locally.
+   
+   ![alt text](./images/InstallTypeSpec_SelectVSIXFileTest.png)
+
+_Option 2_. Install typespec with vscode extension marketplace:
+   `Extension` -> input `TypeSpec for VS Code` -> `Install`
+   
+   ![alt text](./images/InstallTypespec_ExtensionMarketplaceTest01.png)
+
+### Step 2: Trigger generate from TypeSpec
+
+Generation from a TypeSpec can be triggered in two ways:
+
+_Option 1_. Clicking `Generate from TypeSpec` in the _Context Menu_ for a .tsp file in the extended TypeSpec project.
+   
+   ![alt text](./images/TriggerGeneratefromTypeSpec_ContextMenu.png)
+
+_Option 2_. Typing `>TypeSpec: Generate from TypeSpec` in the _Command Palette_ with at least a TypeSpec project folder extended in the _Side Bar_.
+   
+   ![alt text](./images/TriggerGeneratefromTypeSpec_CommandPalette.png)
+
+### Step 3: Click the command `TypeSpec: Generate from TypeSpec`, and choose a project.
+
+   **Validate:** There should be a prompt "Select a Typespec Project".
+
+   ![alt text](./images/GeneratefromTypeSpec_SelectTypespecProject.png)
+
+### Step 4: Select an Emitter Type.
```

Comment:
the UI will be a little different after Compiler can provide template for default emitters. Will keep you updated.